airflow-core/src/airflow/ui/src/components/TriggerDag/TriggerDAGModal.tsx

```diff
@@ -37,22 +45,72 @@ const TriggerDAGModal: React.FC<TriggerDAGModalProps> = ({
   isPaused,
   onClose,
   open,
-}) => (
-  <Dialog.Root lazyMount onOpenChange={onClose} open={open} size="xl" unmountOnExit>
-    <Dialog.Content backdrop>
-      <Dialog.Header paddingBottom={0}>
-        <VStack align="start" gap={4}>
-          <Heading size="xl">Trigger Dag - {dagDisplayName}</Heading>
-        </VStack>
-      </Dialog.Header>
-
-      <Dialog.CloseTrigger />
-
-      <Dialog.Body>
-        <TriggerDAGForm dagId={dagId} isPaused={isPaused} onClose={onClose} open={open} />
-      </Dialog.Body>
-    </Dialog.Content>
-  </Dialog.Root>
-);
+}) => {
+  const [runMode, setRunMode] = useState<RunMode>(RunMode.SINGLE);
+  const { data: dag } = useDagServiceGetDag(
+    {
+      dagId,
+    },
+    undefined,
+    {
+      enabled: open && runMode === RunMode.BACKFILL,
+    },
+  );
+
+  const hasSchedule = dag?.timetable_summary !== null;
+
+  useEffect(() => {
```

Comment:
@bbovenzi / @pierrejeambrun I remember that using `useEffect()` is somehow "bad practice" - Maybe I am lagging a bit of understanding - is the use here OK or shall a different construct be used?